5-3_data_wrangling_xml/data_wrangling_xml.py

```diff
@@ -0,0 +1,211 @@
+
+# Create a top 10 DataFrame and adjust index to match ranking
+top_10_ethnic_groups = df_ethnic_populations.head(10)
+top_10_ethnic_groups.index = range(1,11)
+
+# Round population values to whole numbers
+top_10_ethnic_groups.population = top_10_ethnic_groups.population.apply(floor)
+
+
+# -----------------------------------------------------
+# Name and country of a) longest river, b) largest lake, and c) airport at highest elevation
+
+# Here we create a 'countries' dictionary to help map country_code information to the actual country name.
+# This will be used on each of the three parts of this question
+countries = {}
+for country in document.iterfind('country'):
+    countries[country.attrib['car_code']] = country.find('name').text
+
+#A
+rivers = []
+for river in document.iterfind('river'):
+    name = river.find('name').text
+    # Determine Location
+    if river.find('located') is not None:
+        country = countries[river.find('located').attrib['country']]
+    else:
+        country = countries[river.find('source').attrib['country']]
+
+    # Determine Length
+    if river.find('length') is not None:
+        length = float(river.find('length').text)
+    else:
+        length = None
+
+    # Add dictionary of answers to 'rivers' list
+    rivers.append({
+        'name': name,
+        'country': country,
+        'length': length
+    })
+
+df_rivers = pd.DataFrame.from_records(rivers).sort_values('length',ascending=False).reset_index(drop=True)
+longest_river = df_rivers.iloc[0]
+
+
+#B
+lakes = []
+for lake in document.iterfind('lake'):
+    name = lake.find('name').text
+    # Determine Location
+    if lake.find('located') is not None:
+        country = countries[lake.find('located').attrib['country']]
+    else:
+        country = countries[lake.attrib['country'].split(' ')[0]]
+
+    # Determine size
+    if lake.find('area') is not None:
+        area = float(lake.find('area').text)
+    else:
+        area = None
+
+    # Add dictionary of answers to 'lakes' list
+    lakes.append({
+        'name': name,
+        'country': country,
+        'area': area
+    })
+
+df_lakes = pd.DataFrame.from_records(lakes).sort_values('area',ascending=False).reset_index(drop=True)
+largest_lake = df_lakes.iloc[0]
+
+
+#C
+airports = []
+for airport in document.iterfind('airport'):
+    name = airport.find('name').text
+    country = countries[airport.attrib['country']]
+
+    # Determine elevation
+    if airport.find('elevation').text is not None:
+        elevation = float(airport.find('elevation').text)
+    else:
+        elevation = None
+
+    # Add dictionary of answers to 'airports' list
+    airports.append({
+        'name': name,
+        'country': country,
+        'elevation': elevation
+    })
+
+df_airports = pd.DataFrame.from_records(airports).sort_values('elevation',ascending=False).reset_index(drop=True)
+highest_airport = df_airports.iloc[0]
```

Comment:
Newline at the end of files http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline